client/src/models/Task.js

```diff
@@ -1,9 +1,25 @@
 import { Settings } from "api"
-import Model, { createYupObjectShape, yupDate } from "components/Model"
+import Model, {
+  createYupObjectShape,
+  NOTE_TYPE,
+  yupDate
+} from "components/Model"
+import _isEmpty from "lodash/isEmpty"
+import { Report } from "models"
 import TASKS_ICON from "resources/tasks.png"
 import utils from "utils"
 import * as yup from "yup"
 
+function createTaskAssessmentSchema(customFieldsConfig) {
+  const taskAssessmentSchemaShape = createYupObjectShape(
+    customFieldsConfig,
+    "entityAssessment"
+  )
+  return yup.object().shape({
+    entityAssessment: taskAssessmentSchemaShape
```

Comment:
Then this would become `[ENTITY_ASSESSMENT_FIELD]: taskAssessmentSchemaShape`.

Reply:
See above, I will do that as part of #2954